### PR TITLE
feat: Function Refactor Part 0: Functions are just facades for model logic

### PR DESCRIPTION
**Pull request by kristofbolyai**
Removes `ActiveFunction`. There should be no case where a function needs to be disabled, enabled or have any kind of event logic. All of the above should be kept in the respective model. I kept crash protection, if for whatever reason we still crash (likely a formatting error).

### CODE REVIEW

**Comment by kristofbolyai** [comment]
This should be merged before the whole refactor, to have this structure going on main.

File: common/src/main/java/com/wynntils/models/experience/CombatXpModel.java [hunk 1]
```diff
@@ -38,10 +40,19 @@ public class CombatXpModel extends Model {
 
     private boolean firstJoinHappened = false;
 
+    private final TimedSet<Float> lastMinuteXpGainRawSet = new TimedSet<>(1, TimeUnit.MINUTES, true);
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/experience/CombatXpModel.java]: I understand what you mean, but "last minute" sounds like it is almost too late, [cf](https://www.dictionary.com/browse/at-the-last-minute).

Maybe "latestMinute"? Maybe we need assistance from a native English speaker. @P0keDev ?

**Review by magicus — DISMISSED** [review]
Change itself look good but we need better name for "last minute".

File: common/src/main/java/com/wynntils/models/experience/CombatXpModel.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/experience/CombatXpModel.java]: @magicus I renamed it. Do you accept it so I can merge this (and come back to it if needed)?

**Review by magicus — APPROVED** [review]

**Pull request merged**